lib/core/CJsonStateRestoreTraverser.cc

```diff
@@ -223,15 +228,58 @@ const std::string& CJsonStateRestoreTraverser::nextValue() const {
 }
 
 bool CJsonStateRestoreTraverser::parseNext(bool remember) {
-    if (m_Reader.HasParseError()) {
-        this->logError();
-        return false;
-    }
+    bool ret{true};
+    SBoostJsonHandler::ETokenType currentTokenType = m_Handler.s_Type;
+    do {
+        if (m_Reader.last_error()) {
+            this->logError();
+            ret = false;
+            break;
+        }
+
+        m_Handler.s_RememberValue = remember;
+
+        char c = '\0';
+        m_ReadStream.get(c);
+        if (c == '\0') {
+            break;
+        }
```

Comment:
For efficiency it would be best to read, say, 256 characters into a stack buffer and use `memchr` to check for zero bytes within it. Reading one character at a time is very inefficient.